### PR TITLE
[MRG] FIX: tstop cleanup

### PR DESCRIPTION
**Pull request by jasmainak**
Follow up #397 so I don't forget. Not critical for 0.2 release

Also need to remove:

https://github.com/jonescompneurolab/hnn-core/blob/af26624dac0fd495575e627420a60b7d5b403b67/hnn_core/drives.py#L144-L145

as suggested in [this comment](https://github.com/jonescompneurolab/hnn-core/pull/397#discussion_r678024474)

### CODE REVIEW

**Comment by jasmainak** [comment]
okay this is good to go for me, I'll task switch tomorrow so folks can work peacefully without rebase issues :)

File: hnn_core/network_builder.py [hunk 1]
```diff
@@ -340,17 +340,6 @@ def _build(self):
         if _get_rank() == 0:
             print('[Done]')
 
-    def __enter__(self):
```
Comment (cjayb) [inline hnn_core/network_builder.py:343]: Why aren't these needed anymore?

File: hnn_core/parallel_backends.py [hunk 2]
```diff
@@ -679,7 +690,7 @@ def simulate(self, net, n_trials, postproc=False):
         env = _get_mpi_env()
 
         self.proc, sim_data = run_subprocess(
-            command=self.mpi_cmd, obj=net, timeout=30,
+            command=self.mpi_cmd, obj=[net, tstop, dt, n_trials], timeout=30,
```
Comment (cjayb) [inline hnn_core/parallel_backends.py:693]: Cool! `obj` is a list of objects!

File: hnn_core/parallel_backends.py [hunk 3]
```diff
@@ -518,6 +518,10 @@ def simulate(self, net, n_trials, postproc=False):
             connected.
         n_trials : int
             Number of trials to simulate.
+        tstop : float
+            The simulation stop time (ms).
+        dt : float
+            The integration time step of h.CVode (s)
```
Comment (cjayb) [inline hnn_core/parallel_backends.py:524]: This is not correct: the integration time step is given in _milliseconds_, just like the rest of our times. This needs to be correct in three places (two here and one in `dipole.py`.

File: hnn_core/tests/conftest.py [hunk 4]
```diff
@@ -115,5 +111,9 @@ def _run_hnn_core_fixture(backend=None, n_procs=None, n_jobs=1,
                                    record_isoma=record_vsoma,
                                    postproc=postproc, tstop=tstop)
 
+        # number of trials simulated
```
Comment (cjayb) [inline hnn_core/tests/conftest.py:114]: Nice, this is where this test belongs.

**Review by cjayb — APPROVED** [review]
Great improvement @jasmainak , thank you! Just one mistake in the docstring (s -> ms). Feel free to self-merge when ready

File: hnn_core/network_builder.py (hunk 1 above)
Comment (jasmainak) [reply hnn_core/network_builder.py:343]: I think if you go back in the history of `hnn-core` code, I had the idea to do something like this:

```py
with NetworkBuilder(net) as neuron_net:
    simulate_dipole(neuron_net)
```

so you ensure everything gets cleared when `__exit__` is called. I think Blake rightly argued that `NetworkBuilder` should not be visible to the user.

File: hnn_core/parallel_backends.py (hunk 3 above)
Comment (jasmainak) [reply hnn_core/parallel_backends.py:524]: good catch! I don't know why in my head I was equating this with the interval at which the print outputs were happening :)

**Comment by jasmainak** [comment]
Self-merged as requested by @cjayb . Thanks for the comments!

**Pull request merged**